Sources/Filters.swift

```diff
@@ -89,6 +89,44 @@ struct StringFilters {
     }
   }
 
+  /// Converts camelCase to snake_case. Takes an optional Bool argument for making the string lower case,
+  /// which defaults to true
+  /// - parameter value: the value to be processed
+  /// - parameter arguments: the arguments to the function, expecting zero or one argument
+  /// - returns: the snake case string
+  /// - throws: FilterError.invalidInputType if the value parameter isn't a string
+  static func camelToSnakeCase(_ value: Any?, arguments: [Any?]) throws -> Any? {
+    let toLower = parseBool(from: arguments, index: 0) ?? true
+    guard let string = value as? String else { throw FilterError.invalidInputType }
+
+    let snakeCase = try snakecase(string)
+    if toLower {
+      return snakeCase.lowercased()
+    }
+    return snakeCase
+  }
+
+  /// Parses filter arguments for a boolean value, where true can by any one of: "true", "yes", "1", and
+  /// false can be any one of: "false", "no", "0".
+  /// - parameter arguments: an array of argument values, may be empty
+  /// - parameter index: the index in the arguments array
+  /// - returns: true or false if a value was parsed, or nil if it wasn't able to
+  static func parseBool(from arguments: [Any?], index: Int) -> Bool? {
+    guard index + 1 <= arguments.count else {
+      return nil
+    }
+
+    let boolArg = (arguments[index] as? String ?? "").lowercased()
+    switch boolArg {
+    case "false", "no", "0":
+      return false
+    case "true", "yes", "1":
+      return true
+    default:
+      return nil
```

Comment:
Hmm. I'll refactor. If it's an optional argument (which I tested that Stencil allows for it), it should return nil, otherwise it should throw.